File: lib/Notification/Subscription.dart
// Observer Pattern
class Subscription {
  String clientName;
  List<String> notificationTypes = [];

  Subscription(this.clientName);

  void addNotificationType(String type) {
    notificationTypes.add(type);
  }

  void showSubscribedTypes() {
    print("$clientName is subscribed to: ${notificationTypes.join(', ')}");
  }
}

File: src/Notification/Subscription.java
package Notification;

import java.util.ArrayList;
import java.util.List;


public class Subscription {
    private String clientName;
    private List<String> notificationTypes = new ArrayList<>();

    public Subscription(String clientName) {
        this.clientName = clientName;
    }

    public void addNotificationType(String type) {
        notificationTypes.add(type);
    }

    public void showSubscribedTypes() {
        System.out.println(getClientName() + " is subscribed to: " + String.join(", ", getNotificationTypes()));
    }

    public String getClientName() {
        return clientName;
    }

    public List<String> getNotificationTypes() {
        return notificationTypes;
    }
}
